src/ucp/core/ucp_mm.c

```diff
@@ -102,24 +102,25 @@ ucs_status_t ucp_mem_rereg_mds(ucp_context_h context, ucp_md_map_t reg_md_map,
             ucs_assert(alloc_md_memh_p != NULL);
             uct_memh[memh_index++] = *alloc_md_memh_p;
             new_md_map            |= UCS_BIT(md_index);
-        } else if (!length) {
+        } else if (length == 0) {
             /* don't register zero-length regions */
             continue;
         } else if (md_attr->cap.flags & UCT_MD_FLAG_REG) {
-            if (!(md_attr->cap.reg_mem_types & UCS_BIT(mem_type))) {
-                status = UCS_ERR_UNSUPPORTED;
-            } else {
-                ucs_assert(address && length);
+            ucs_assert(address != NULL);
+            ucs_assert(length != 0);
```

Comment:
this assert is excessive, we could not get to this branch if `length == 0`

Reply:
fixed